databricks-sdk-java/src/main/java/com/databricks/sdk/core/oauth/Token.java

```diff
@@ -21,28 +23,51 @@ public class Token {
    */
   @JsonProperty private LocalDateTime expiry;
 
+  private final ClockSupplier clockSupplier;
+
   /** Constructor for non-refreshable tokens (e.g. M2M). */
   public Token(String accessToken, String tokenType, LocalDateTime expiry) {
-    this(accessToken, tokenType, null, expiry);
+    this(accessToken, tokenType, null, expiry, null);
```

Comment:
Slightly simpler would be to pass `new RealClockSupplier()` here, then you don't need an if condition in the other case, and you can use `Objects.requireNonNull`.

Reply:
@mgyucht thanks for review. That would be bad CUJ for someone calling the other constructor right? They would have to pass Clock Supplier which outside of test should always be System Clock. 

The clock shouldn't be exposed on customer end I think.

Reply:
Isn't the other constructor only used for tests? If you want to use the system clock, you should use this constructor.

Reply:
Right, I was thinking if someone wants to pass null in clockSupplier -- we should have a case for that but they should then use the other constructor. Updated.